### PR TITLE
feat: NaN handling across signal_utils; plot_dff overhaul; ruff migration; MkDocs

### PR DESCRIPTION
**Pull request by j-friedrich**
## Summary

Accumulated changes from `dev` since v0.2.0:

**Signal utils (NaN handling):**
- `percentile_filter`: always uses `scipy.ndimage.percentile_filter` (O(log n) since scipy ≥ 1.15.0); adds `skipna=False` for NaN-aware filtering via pandas rolling quantile; fix `size==1` + `skipna=True` returning empty array
- `median_filter`: adds `skipna=False`, delegates to `percentile_filter`
- `fill_nan`: renamed from `_fill_nan` (now public); returns copy unchanged on all-NaN input
- `nanmedian_filter`: deprecated with `DeprecationWarning`; use `median_filter(..., skipna=True)` instead; fix return type annotation `np.array` → `np.ndarray`
- `robust_std`: adds `skipna=False`
- `noise_std`: `skipna=True` works for all three methods (`mad`, `fft`, `welch`); fft+skipna 2D recursion now correctly forwards `device`, `noise_range`, and `max_num_samples`

**dff:**
- `_dff_single_trace`: uses `median_filter(skipna=True)` + `fill_nan` instead of deprecated `nanmedian_filter`
- `plot_dff`: overhauled aesthetics — distinct colors per quantity, `\mathrm` mathtext labels, frameless legend, `zorder=-1` on dF/F traces, tuned figsize/margins to match capsule output dimensions; new 4-panel full-baseline mode via `F0trend` parameter

**summary_images:**
- Fix `torch.mean`/`torch.nanmean` calls using `axis=` instead of `dim=`

**motion_border_utils:**
- Separate `frame_shift` result variable from numeric `max_shift` threshold to fix misleading error message
- Fix spelling: "analomies" → "anomalies"

**array_utils:**
- Wrap `h5py.File` opens in `_downsample_group` and `_subsample_group` with context managers to prevent file handle leaks under parallel use
- Switch to `multiprocessing.get_context("fork").Pool` (falls back to spawn when fork unavailable) — fixes macOS spawn-vs-fork locking failures that caused 6 `test_downsample_compressed_h5` tests to fail

**Tooling** (per [AIND software standards](https://docs.allenneuraldynamics.org/en/latest/policies_practices/software_practices.html#standards-tools)):
- Replace `black`, `flake8`, `isort` with `ruff format` + `ruff check`
- CI now also runs on PRs targeting `dev`
- Migrate docs from Sphinx/furo to MkDocs with mkdocstrings (numpy style, material theme)
- Remove `doc_template/`; add `mkdocs.yml` and `docs/`

**Style:**
- Collapse multiline expressions unnecessarily split by Black's trailing commas that fit within the 100-char limit (`baseline_fitting.py`, `dff.py`, `summary_images.py`)

## Test plan

- [ ] All existing tests pass
- [ ] New tests cover `skipna` paths for all functions, NaN-block edge cases, 2D inputs, and all-NaN guards
- [ ] Added `test_noise_std_axis` covering the `axis != -1` moveaxis branch
- [ ] Removed redundant `test_noise_std_skipna_methods` (covered by `test_noise_std_nan` + `test_noise_std_skipna_2d`)
- [ ] `ruff format --check` and `ruff check` and `interrogate` pass on changed files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by j-friedrich** [comment]
Addressing the two Copilot review comments from https://github.com/AllenNeuralDynamics/aind-ophys-utils/pull/72#pullrequestreview-4665850188:

- **`fill_nan` all-NaN crash**: fixed — now checks `no_nan_indices.size == 0` and returns a copy unchanged before calling `np.interp`.
- **`noise_std` fft+skipna 2D parameter forwarding**: fixed — the per-row recursion now correctly forwards `max_num_samples`, `noise_range`, and `device`.

Both fixes were committed on the `plot-dff-style` branch and are now in `dev` via PR #73.

**Comment by j-friedrich** [comment]
Addressed all three issues from this review in dev (commits `e3b1a70` → `ed31312`):

1. **`_downsample_group` / `_subsample_group` file handle leaks** — wrapped both `h5py.File(...)` calls in context managers and read the slice into a numpy array before closing, so the file handle is released immediately after the read. Also switched from `multiprocessing.Pool` to `multiprocessing.get_context("fork").Pool` — the pre-existing test failures were a macOS spawn-vs-fork issue: spawned workers competed for the HDF5 POSIX lock on the same file. Fork workers inherit the parent's open file descriptor and avoid this entirely. All 6 previously failing `test_downsample_compressed_h5` tests now pass.

2. **`percentile_filter` docstring** — clarified that O(log n) complexity requires scipy ≥ 1.15.0 (the declared minimum is 1.11).

File: src/aind_ophys_utils/array_utils.py [hunk 1]
```diff
@@ -38,24 +35,20 @@ def n_frames_from_hz(
     return max(1, frames_to_group)
 
 
-def _downsample_group(
-    i, h5py_name, h5py_key, factors, fun, dtype=None, cval=np.nan
-):
+def _downsample_group(i, h5py_name, h5py_key, factors, fun, dtype=None, cval=np.nan):
     """Auxiliary function to compute group max/mean/medians in parallel"""
     array = h5py.File(h5py_name)[h5py_key]
```
[the remark this answers is not in the document]
Comment (j-friedrich) [reply src/aind_ophys_utils/array_utils.py]: see https://github.com/AllenNeuralDynamics/aind-ophys-utils/pull/72#issuecomment-4961342938

File: src/aind_ophys_utils/array_utils.py [hunk 2]
```diff
@@ -68,9 +61,7 @@ def _i0(s, f):
     return 0 if s == "first" else (f - 1 if s == "last" else f // 2)
 
 
-def _subsample_group(
-    i, h5py_name, h5py_key, factors, strategy="first", dtype=None
-):
+def _subsample_group(i, h5py_name, h5py_key, factors, strategy="first", dtype=None):
     """Auxiliary function to select first/last/mid of group in parallel"""
     out = h5py.File(h5py_name)[h5py_key][i][
```
[the remark this answers is not in the document]
Comment (j-friedrich) [reply src/aind_ophys_utils/array_utils.py]: see https://github.com/AllenNeuralDynamics/aind-ophys-utils/pull/72#issuecomment-4961342938

File: src/aind_ophys_utils/signal_utils.py [hunk 3]
```diff
@@ -15,11 +16,14 @@ def percentile_filter(
...
 ) -> np.ndarray:
     """
-    Fast 1D running percentile filter using reflection
-    to extend the input array beyond its boundaries.
-    Uses pandas if input and filter size are long, scipy if short.
+    Fast 1D running percentile filter with reflect boundary handling.
+
+    Uses :func:`scipy.ndimage.percentile_filter` which has O(log n) complexity
+    since scipy 1.15.0. When ``skipna=True``, falls back to a pandas rolling
+    quantile which ignores NaN values within each window.
```
[the remark this answers is not in the document]
Comment (j-friedrich) [reply src/aind_ophys_utils/signal_utils.py]: see https://github.com/AllenNeuralDynamics/aind-ophys-utils/pull/72#issuecomment-4961342938

File: src/aind_ophys_utils/array_utils.py [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 from itertools import product
-from multiprocessing.pool import Pool, ThreadPool
-
+from multiprocessing.pool import ThreadPool
 
 import h5py
 import numpy as np
 from skimage.measure import block_reduce
 
+Pool = multiprocessing.get_context("fork").Pool
```
[the remark this answers is not in the document]
Comment (j-friedrich) [reply src/aind_ophys_utils/array_utils.py]: committed your suggestion
